spec/tbd_tests_spec.rb

```diff
@@ -1235,6 +1237,93 @@
 
     file = File.join(__dir__, "files/osms/out/seb_KIVA4.osm")
     model.save(file, true)
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # Recover KIVA-populated model and re- set/gen KIVA.
+    argh              = {}
+    argh[:option    ] = "(non thermal bridging)"
+    argh[:gen_kiva  ] = true
+    argh[:reset_kiva] = true
+
+    json     = TBD.process(model, argh)
+    expect(json).to be_a(Hash)
+    expect(json).to have_key(:io)
+    expect(json).to have_key(:surfaces)
+    io       = json[:io      ]
+    surfaces = json[:surfaces]
+    expect(TBD.info?).to be true
+    expect(TBD.logs.size).to eq(1)
+    expect(TBD.logs.first[:message]).to include("Purged KIVA objects from ")
+    expect(surfaces).to be_a(Hash)
+    expect(surfaces.size).to eq(56)
+    expect(io).to be_a(Hash)
+    expect(io).to have_key(:edges)
+    expect(io[:edges].size).to eq(106)
+
+    slabs = 0
+
+    # Same outcome as "seb_KIVA4.osm".
+    surfaces.each do |id, s|
+      next unless s.key?(:kiva)
+
+      slabs += 1
+      expect(s).to have_key(:exposed)
+      slab = model.getSurfaceByName(id)
+      expect(slab).to_not be_empty
+      slab = slab.get
+
+      expect(slab.adjacentFoundation).to_not be_empty
+      perimeter = slab.surfacePropertyExposedFoundationPerimeter
+      expect(perimeter).to_not be_empty
+      perimeter = perimeter.get
+
+      per = perimeter.totalExposedPerimeter
+      expect(per).to_not be_empty
+      per = per.get
+      expect((per - s[:exposed]).abs).to be_within(TOL).of(0)
+
+      expect(per).to be_within(TOL).of( 8.81) if id == "Small office 1 Floor"
+      expect(per).to be_within(TOL).of( 8.21) if id == "Utility 1 Floor"
+      expect(per).to be_within(TOL).of(12.59) if id == "Open area 1 Floor"
+      expect(per).to be_within(TOL).of( 6.95) if id == "Entry way  Floor"
+    end
+
+    expect(slabs).to eq(4)
+
+    # Forward-translating/running either "seb_KIVA4.osm" or "seb_KIVA5.osm"
+    # would yield the same simulation results.
+    file = File.join(__dir__, "files/osms/out/seb_KIVA5.osm")
```

Comment:
Sanity check to ensure very same outcome between:

 - a model with default TBD-generated KIVA inputs
 - same model ... first purged of existing KIVA inputs, then provided with default TBD-generated KIVA inputs